ui/appui-react/src/appui-react/ui-items-provider/UiItemsProvider.ts

```diff
@@ -24,21 +24,59 @@ export interface UiItemsProvider {
   /** Id of provider. */
   readonly id: string;
 
-  /** Provides toolbar items. */
+  /** Describes stages for which the items should be provided.
+   * @note Ignored for `provideToolbarItems`, `provideStatusBarItems`, `provideBackstageItems`, `provideWidgets` methods.
+   * @alpha
+   */
+  readonly stageIds?: ReadonlyArray<string>;
```

Comment:
These are "ands" between the arrays, i.e. if `Provider1` is allowed in `Frontstage1` and `Usage1` it will not load items to `Frontstage2` `Usage1` or `Frontstage1` `Usage2`.

Reply:
With this change, we aim to group the providers not only features but also frontstages.
The problem with a separate array is that it:
- Requires twice as many APIs and overrides.
- Enables "global" providers which provide items to multiple stages. This complicates overriding, i.e. need to know item IDs to map correctly or other implementation detail...

Reply:
How is overriding working with this API ? If I create a measure tool provider, I will probably not put the stage id and stage usage info, how will an application decide where to put things ? #91

Reply:
The application would override the `stageIds` property of a provider when registering.